src/Tasks/FileIO/GetFileHash.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Computes the checksum for a single file.
+    /// </summary>
+    public sealed class GetFileHash : TaskExtension
+    {
+        internal const string DefaultFileHashAlgorithm = "SHA256";
+
+        /// <summary>
+        /// The files to be hashed.
+        /// </summary>
+        [Required]
+        public ITaskItem[] Files { get; set; }
+
+        /// <summary>
+        /// The algorithm. Allowed values: SHA256, SHA384, SHA512. Default = SHA256.
+        /// </summary>
+        public string Algorithm { get; set; } = DefaultFileHashAlgorithm;
+
+        /// <summary>
+        /// The metadata name where the hash is store in each item. File hash is in hex. Defaults to "FileHash".
+        /// </summary>
+        public string MetadataName { get; set; } = "FileHash";
+
+        /// <summary>
+        /// The metadata name where the base64 encoded hash is store in each item. File hash is in hex. Defaults to "FileHashBase64".
+        /// </summary>
+        public string MetadataNameBase64 { get; set; } = "FileHashBase64";
+
+        /// <summary>
+        /// The hash of the file in hex. This is only set if there was one item group passed in.
+        /// </summary>
+        [Output]
+        public string Hash { get; set; }
+
+        /// <summary>
+        /// The hash of the file base64 encoded. This is only set if there was one item group passed in.
+        /// </summary>
+        [Output]
+        public string HashBase64 { get; set; }
+
+        /// <summary>
+        /// The input files with additional metadata set to include the file hash.
+        /// </summary>
+        [Output]
+        public ITaskItem[] Items { get; set; }
+
+        public override bool Execute()
+        {
+            if (!SupportsAlgorithm(Algorithm))
+            {
+                Log.LogErrorFromResources("FileHash.UnrecognizedHashAlgorithm", Algorithm);
+                return false;
+            }
+
+            foreach (var file in Files)
+            {
+                if (!File.Exists(file.ItemSpec))
+                {
+                    Log.LogErrorFromResources("FileHash.FileNotFound", file.ItemSpec);
+                    continue;
+                }
+
+                var hash = ComputeHash(Algorithm, file.ItemSpec);
+                file.SetMetadata("FileHashAlgoritm", Algorithm);
+                file.SetMetadata(MetadataName, ConversionUtilities.ConvertByteArrayToHex(hash));
```

Comment:
Should the user be able to choose which of the two representations to use? I assume that they would only use one of them, so the other becomes wasted memory.

Reply:
Would it be better if we changed the tasks to support a property such as `UseBase64Encoding="true"` instead? That would allow us to operating in hex or base64 mode and avoid wasting memory.

Reply:
Maybe an input property called "HashEncoding" with two valid options, `hex` and `base64`? And then single metadata / output would be either one or the other.